textBasedAdventure.js

```diff
@@ -1,355 +1,412 @@
+    chances--;
+    if(chances === 0){
+      console.log("No more chances, have a good life!")
+      leaveGame();
     }
-        if (first === "true") {
-            while (second !== "true" && second !== "false") {
-            second = readlineSync.question("12B. Is the sky blue? \n");
-            console.log(" ");
-            second = second.trim().toLowerCase();
-                if (second !== "true" && second !== "false") {
-                    console.log(" ");
-                    console.log("Oh man , that was an easy one.");
-                    console.log(" ");
-                }
-            }
-            if (second === "true")
-            console.log(" ");
-            console.log("You won! Is your IQ 2000 or something?");
-            console.log(" ");
-            playGameAgain();
-        } else {
-            // chances--
-            console.log("You lose! Keep trying!");
-            console.log(" ");
-            playGameAgain();
-        }
+    console.log(`\n You have ${chances} chances left!\n`);
+    answer = readlineSync.keyInYN(
+      "\n That's incorrect, do you want to try again?\n"
+    );
+    if (answer === true) {
+      isQuestionTwelve();
+    }
+  }
 };
 
 // // Question Eleven
 const questionEleven = () => {
   let impeachedPres;
-  let chances;
   while (impeachedPres !== "3" && impeachedPres !== "three") {
     impeachedPres = readlineSync.question(
-      "11. How many presidents have been impeached? \n"
+      "\n 11. How many presidents have been impeached? \n"
     );
     impeachedPres = impeachedPres.trim().toLowerCase();
     if (impeachedPres !== "3" && impeachedPres !== "three") {
-        console.log(" ");
-      console.log("I know , i know who cares");
-      console.log(" ");
+      console.log("\n I know , i know who cares\n");
+      chances--;
+      if(chances === 0){
+        console.log("No more chances, have a good life!")
+        leaveGame();
+      }
+      console.log(`\n You have ${chances} chances left!\n`);
+      answer = readlineSync.keyInYN(
+        "\n That's incorrect, do you want to try again?\n"
+      );
+      if (answer === true) {
+        questionEleven();
+      }
+    }
+    if (impeachedPres === "3" || impeachedPres === "three") {
+      isQuestionTwelve();
+    } else {
+      isQuestionTwelve();
     }
-  }
-  if (impeachedPres === "3" || impeachedPres === "three") {
-      console.log(" ");
-    isQuestionTwelve();
-  } else {
-    // chances--
-    isQuestionEleven();
   }
 };
 
 // // Question Ten
 const questionTen = () => {
   let cartoonCharacter;
-  let chances;
   while (cartoonCharacter !== "sponge bob") {
     cartoonCharacter = readlineSync.question(
-      "10. Which cartoon character lives in a pineapple under the sea? \n"
+      "\n 10. Which cartoon character lives in a pineapple under the sea? \n"
     );
     cartoonCharacter = cartoonCharacter.trim().toLowerCase();
     if (cartoonCharacter !== "sponge bob") {
```

Comment:
What if the player includes "square pants" in their answer or they write "spongebob" with no space in between? -- How can you consider these possible responses in your project?